src/app/pages/system/general-settings/ntp-server/ntp-server-list/ntp-server-list.component.spec.ts

```diff
@@ -0,0 +1,167 @@
+    prefer: false,
+    minpoll: 6,
+    maxpoll: 10,
+  },
+  {
+    id: 3,
+    address: '1.debian.pool.ntp.org',
+    burst: false,
+    iburst: true,
+    prefer: false,
+    minpoll: 6,
+    maxpoll: 10,
+  },
+  {
+    id: 10,
+    address: '0.debian.pool.ntp.org',
+    burst: false,
+    iburst: true,
+    prefer: false,
+    minpoll: 6,
+    maxpoll: 10,
+  },
+];
+
+describe('NtpServerListComponent', () => {
+  let spectator: Spectator<NtpServerListComponent>;
+  let loader: HarnessLoader;
+  let ws: WebSocketService;
+
+  const createComponent = createComponentFactory({
+    component: NtpServerListComponent,
+    imports: [
+      EntityModule,
+      IxTableModule,
+    ],
+    providers: [
+      mockProvider(DialogService),
+      mockProvider(IxSlideInService),
+      mockWebsocket([
+        mockCall('system.ntpserver.query', fakeDataSource),
+        mockCall('system.ntpserver.delete'),
+      ]),
+    ],
+  });
+
+  beforeEach(() => {
+    spectator = createComponent();
+    loader = TestbedHarnessEnvironment.loader(spectator.fixture);
+    ws = spectator.inject(WebSocketService);
+    jest.spyOn(spectator.fixture.componentInstance, 'doAdd').mockImplementation();
+    jest.spyOn(spectator.fixture.componentInstance, 'doDelete').mockImplementation();
+    spectator.fixture.detectChanges();
+  });
+
+  afterEach(() => {
+    jest.restoreAllMocks();
+  });
+
+  it('should show table headers', async () => {
+    const table: IxTableHarness = await loader.getHarness(IxTableHarness);
+    const headerRow = await table.getHeaderRow();
+
+    expect(Object.keys(headerRow).length).toBe(7);
+    expect(headerRow.address).toBe('Address');
+    expect(headerRow.burst).toBe('Burst');
+    expect(headerRow.iburst).toBe('IBurst');
+    expect(headerRow.prefer).toBe('Prefer');
+    expect(headerRow.minpoll).toBe('Min Poll');
+    expect(headerRow.maxpoll).toBe('Max Poll');
+    expect(headerRow.actions).toBe('');
+  });
+
+  it('should show table rows', async () => {
+    spectator.fixture.componentInstance.loading = false;
+    spectator.fixture.componentInstance.error = false;
+    spectator.fixture.componentInstance.createDataSource(fakeDataSource);
+    const table = await loader.getHarness<IxTableHarness>(IxTableHarness);
+    const cells = await table.getRowCells();
+
+    const expectedRows = [
+      {
+        actions: 'delete',
+        address: '2.debian.pool.ntp.org',
+        burst: 'false',
+        iburst: 'true',
+        prefer: 'false',
+        minpoll: '6',
+        maxpoll: '10',
+      },
+      {
+        actions: 'delete',
+        address: '1.debian.pool.ntp.org',
+        burst: 'false',
+        iburst: 'true',
+        prefer: 'false',
+        minpoll: '6',
+        maxpoll: '10',
+      },
+      {
+        actions: 'delete',
+        address: '0.debian.pool.ntp.org',
+        burst: 'false',
+        iburst: 'true',
+        prefer: 'false',
+        minpoll: '6',
+        maxpoll: '10',
+      },
+    ];
```

Comment:
I guess my question is whether we care about column names in this case, since they are not shown to the user.
How about an even simplier method:
```
const cells = await table.getCells();
expect(cells).toMatchObject([
 ['Address', 'Burst', 'IBurst', ...],
 ['2.debian.pool.ntp.org', 'false', 'true'...],
);
```